fastparquet/test/test_writer.py

```diff
@@ -272,6 +283,23 @@ def test_groups_roundtrip(tempdir):
         assert row.b in list(df[(df.a==row.a)&(df.c==row.c)].b)
 
 
+def test_empty_groupby(tempdir):
+    df = pd.DataFrame({'a': np.random.choice(['a', 'b', None], size=1000),
+                       'b': np.random.randint(0, 64000, size=1000),
+                       'c': np.random.choice([True, False], size=1000)})
+    df.loc[499:, 'c'] = True  # no False in second half
+    writer.write(tempdir, df, partition_on=['a', 'c'], file_scheme='hive',
+                 partitions=[0, 500])
+    r = ParquetFile(tempdir)
+    assert r.count == sum(~df.a.isnull())
+    assert len(r.row_groups) == 6
+    out = r.to_pandas()
```

Comment:
reading partitions
http://spark.apache.org/docs/latest/sql-programming-guide.html#partition-discovery

writing API (called partitionBy)
https://spark.apache.org/docs/1.6.0/api/java/org/apache/spark/sql/DataFrameWriter.html#partitionBy(scala.collection.Seq)

That's just spark, but this is a hive thing originally, I think. For each input split (i.e., separate CSV file or whatever), you get one file from one worker in each possible groupby combination.